run_analysis.py

```diff
@@ -78,81 +87,214 @@ def run_fit(i):
+def cut_variaion(flow_config, outdir, correlated, combined=False, operations=None, batch=False):
+	check_dir(f"{outdir}/cutVar")
+	if correlated:
+		logger("Cut variation will be performed", level="INFO")
+
+		ry_path = f"{outdir}/raw_yields"
+		eff_path = f"{outdir}/eff"
+		method = "--correlated" if correlated else ""
+
+		cmd = (
+			f"python3 {paths['CutVariation']} {flow_config} {ry_path} {eff_path} {method}"
+		)
+		if batch:
+			cmd += " --batch"
+		logger(f"{cmd}", level="COMMAND")
+		os.system(cmd)
+
+	if combined:
+		outdirCorr = os.path.join(os.path.dirname(outdir), os.path.basename(outdir).replace("_combined", "_correlated"))
+		if os.path.exists(f"{outdirCorr}/cutVar/cutVar.root"):
+			logger("Cut variation with correlated cuts found!", level="INFO")
+			cmd = (
+				f"cp {outdirCorr}/cutVar/cutVar.root {outdir}/cutVar/cutVar.root"
+			)
+			logger(f"{cmd}", level="COMMAND")
+			os.system(cmd)
+		else:
+			logger(f"Cut variation with correlated cuts not found in {outdirCorr}, running correlated cut variation", level="WARNING")
+
+			logger("Running analysis with correlated cut variation", level="WARNING")
+			logger("========================================================================================", level="WARNING")
+			import copy
+			operationsCorr = copy.deepcopy(operations)
+			for key in operationsCorr:
+				if key in ['do_cut_variation', 'make_yaml', 'proj_mc', 'proj_data', 'efficiencies', 'get_vn_vs_mass']:
+					operationsCorr[key] = True
+				else:
+					operationsCorr[key] = False
+			os.makedirs(os.path.join(outdir, "config_flow"), exist_ok=True)
+			flow_configCorr = os.path.join(outdir, "config_flow", f"corr_{os.path.basename(flow_config)}")
+			with open(flow_config, 'r') as cfgFlow:
+				config = yaml.safe_load(cfgFlow)
+				config['operations'] = operationsCorr
+				config['outdir'] = outdirCorr #! make sure all output directories are set from function arguments not from config file
+				with open(flow_configCorr, 'w') as cfgFlow:
+					yaml.dump(config, cfgFlow, default_flow_style=True)
+			logger(f"Using temporary flow config: {flow_configCorr}", level="WARNING")
+			run_correlated_cut_variation(flow_configCorr, operationsCorr, nworkers, outdirCorr)
+			logger("========================================================================================", level="WARNING")
+
+			logger("Copying cut variation with correlated cuts to combined cut variation", level="WARNING")
+			cmd = (
+				f"cp {outdirCorr}/cutVar/cutVar.root {outdir}/cutVar/cutVar.root"
+			)
+			logger(f"{cmd}", level="COMMAND")
+			os.system(cmd)
+
+def data_driven_fraction(outdir, syst_frac, batch=False):
+	logger("Data driven fraction will be performed", level="INFO")
+	check_dir(f"{outdir}/frac")
+ 
+	cutvar_file = f"{outdir}/cutVar/cutVar.root"
+	eff_path = f"{outdir}/eff"
+
+	cmd = (
+		f"python3 {paths['DataDrivenFraction']} {cutvar_file} {eff_path}"
+	)
+	if syst_frac:
```

Comment:
I would not put here the piece of code for the systematics, but I would use a dedicated script. Also, we do not need a systematic option but just the possibility to provide the cutvar_file path so I would just remove this argument